test/banned_deps_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+var bannedImports = []string{
+	"github.com/knative",
+	"github.com/kubernetes",
+	"github.com/istio",
+}
+
+func TestBannedImports(t *testing.T) {
+	_, filename, _, _ := runtime.Caller(0)
+	dir := filepath.Dir(filename)
+	dir = fmt.Sprintf("%s%s..%s%s", dir, string(filepath.Separator), string(filepath.Separator), "vendor")
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			return nil
+		}
+		for _, banned := range bannedImports {
+			if strings.HasSuffix(path, banned) {
+				return fmt.Errorf("%s is a banned import", banned)
```

Comment:
Expand this to be more clear that there's an easy alternative. `%s is a banned import. Use the vanity import path instead.`